From d1015cc2a2e83fb16f2cf96e4c40a279d5fca7be Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Thu, 14 Jan 2021 10:03:51 +0000
Subject: [PATCH] Do not show Unknown [***] for every client connection

Ignore the initial client state change from UNKNOWN to IDLE which was being set
as part of the fix in fb36f22.

Fixes https://github.com/fwupd/fwupd/issues/2766
---
 src/fu-progressbar.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/fu-progressbar.c b/src/fu-progressbar.c
index 9a7378c5..5dd9ff39 100644
--- a/src/fu-progressbar.c
+++ b/src/fu-progressbar.c
@@ -297,6 +297,12 @@ fu_progressbar_update (FuProgressbar *self, FwupdStatus status, guint percentage
 {
 	g_return_if_fail (FU_IS_PROGRESSBAR (self));
 
+	/* ignore initial client connection */
+	if (self->status == FWUPD_STATUS_UNKNOWN && status == FWUPD_STATUS_IDLE) {
+		self->status = status;
+		return;
+	}
+
 	/* use cached value */
 	if (status == FWUPD_STATUS_UNKNOWN)
 		status = self->status;
-- 
2.25.1

